### ISSUE TITLE
Setup turbo for development 

### ISSUE THREAD

**Issue by JeelRajodiya** [json-schema-org/tour#10]
There are some hydration errors occurring when used dev with --turbo. resolve those errors to use --turbo

**Comment by JeelRajodiya** [comment]
https://github.com/hashicorp/next-mdx-remote?tab=readme-ov-file#installation

This may help

**Comment by RonakSurana-2001** [comment]
Is this issue available to solve
Can I work on this issue
Also I want to ask is this repository under hackoctoberfest 2024

**Comment by JeelRajodiya** [comment]
@RonakSurana-2001 Yes the repo is participating under hacktoberfest. You can work on it and open a PR.

**Comment by JeelRajodiya** [comment]
@RonakSurana-2001 I am not sure, but maybe upgrading next.js version can fix this issue. it is worth a try.

**Comment by RonakSurana-2001** [comment]
@JeelRajodiya I added --turbo in package.json dev
![Screenshot 2024-10-03 093352](https://github.com/user-attachments/assets/70150c45-0f90-4757-b786-3c828d39f381)
And I am not seeing any Error.
I have also Added 
![Screenshot 2024-10-03 093523](https://github.com/user-attachments/assets/336e6c34-a652-407e-afb9-0871287adeed)
in next.config.mjs as you suggested. but it is of no use as error is not coming.

> https://github.com/hashicorp/next-mdx-remote?tab=readme-ov-file#installation
> 
> This may help

I have raised a PR you can see the change. #86

**Comment by JeelRajodiya** [comment]
Hello @RonakSurana-2001 , please check your PR. I pushed some changes. Now you will be able to see this error. 

![image](https://github.com/user-attachments/assets/39144f4c-1ee7-4a3f-9321-46dd10610586)

**Comment by JeelRajodiya** [comment]
I found these two issues somewhat relevant to ours. 
https://github.com/vercel/turborepo/issues/6094
https://github.com/vercel/next.js/issues/44653

**Comment by RonakSurana-2001** [comment]
Hmm finding the solution

**Comment by RonakSurana-2001** [comment]
Think that issue is not with --turbo
I removed Providers from layout.tsx the error was gone

**Comment by JeelRajodiya** [comment]
please reflect that in your PR

> Think that issue is not with --turbo I removed Providers from layout.tsx the error was gone

**Comment by RonakSurana-2001** [comment]
Can you please check #88 PR.

**Comment by pavanydg** [comment]
@JeelRajodiya is this issue still open?

**Comment by iamridoydey** [comment]
can I work on it?

**Comment by JeelRajodiya** [comment]
@pavanydg Let's give the issue to @iamridoydey, since he is new.

**Comment by JeelRajodiya** [comment]
Please go ahead @iamridoydey. 
Please note that, the issue might not even have any solution or the it is originated from the library itself. You can research online to figure out the solution of the issue.

**Comment by iamridoydey** [comment]
Thanks @JeelRajodiya . I am looking at it. If I stuck then I will ask you for further help . Thanks for the support

**Comment by JeelRajodiya** [comment]
> Thanks @JeelRajodiya . I am looking at it. If I stuck then I will ask you for further help . Thanks for the support

You can also refer to the conversations in this issue. I have shared some of the related resources.

**Comment by iamridoydey** [comment]
Thanks bro @JeelRajodiya

**Comment by iamridoydey** [comment]
I use d "next-dev": "next dev --turbo", in package.json and then I got some error.

These is the error: 

 ✓ Ready in 1160ms
[0]  ⚠ Webpack is configured while Turbopack is not, which may cause problems.
[0]  ⚠ See instructions if you need to configure Turbopack:
[0]   https://nextjs.org/docs/app/api-reference/next-config-js/turbo


I think the problem here is we are using webpack and also trying to use turbo . This may conflict

**Comment by iamridoydey** [comment]
@JeelRajodiya  do you want to use turbo or turbopack? because both are different .

**Comment by iamridoydey** [comment]
@JeelRajodiya  After using 
    "next-dev": "next dev --turbo",
 on the package.json . I got this error 
 
 Webpack is configured while Turbopack is not, which may cause problems.
[0]  ⚠ See instructions if you need to configure Turbopack:
[0]   https://nextjs.org/docs/app/api-reference/next-config-js/turbo

isn't it conflict with each other?

**Comment by JeelRajodiya** [comment]
@iamridoydey I am not sure either : ) Maybe we need to configure turbopack itself, You can open a PR if you find any solution.

**Comment by iamridoydey** [comment]
@JeelRajodiya still some issues. But i am looking at it

**Comment by iamridoydey** [comment]
> @iamridoydey I am not sure either : ) Maybe we need to configure turbopack itself, You can open a PR if you find any solution.

I am opening a pull request. Check it. It is working fine now.

**Comment by JeelRajodiya** [comment]
@iamridoydey 
You can try searching online to find any similar issues, ([see my previous comment](https://github.com/json-schema-org/tour/issues/10#issuecomment-2390602858))

There might be two possible causes for the issue 
1. Wrong trubopack configuration in our repository 
2. Issue from next.js itself (if it is resolved from their side, it can be resolved by just upgrading next.js) 

There might be any other cause as well, which I might be not aware of.

**Comment by iamridoydey** [comment]
I upgrade nextjs ro 15.0.2 there turbopack work just fine. But get issues from other dependancy. So many dependancy are not working with current verions of the nextjs. They wants nextjs 14. There was the actual problem

**Comment by JeelRajodiya** [comment]
@iamridoydey Can you please share a screenshot of the error. Which dependencies are causing the problem?

**Comment by JeelRajodiya** [comment]
@iamridoydey Can you please share a screenshot of the error. Which dependencies are causing the problem?

**Comment by iamridoydey** [comment]
@JeelRajodiya  I will but I am doing another project. Can you give me some time? Probably in the next day. By the way I am also working on a next js project where I use turbopack not experimental way. many dependancy need to update for this. So we can have a good discussion after I upload all of the issue. Can we?

**Comment by JeelRajodiya** [comment]
@iamridoydey Yes, you can take your time.

**Comment by JeelRajodiya** [comment]
@vtushar06 You can give the issue a try, if you solve it, feel free to open a PR. Since many people have tried solving it, let's consider this issue open without any assignment. No assignment is needed open a PR for this issue.

**Comment by vtushar06** [comment]
@JeelRajodiya okay sure, thanks for this green flag, I will update as soon as possible.

**Comment by singhaditya73** [comment]
@JeelRajodiya
Hi, I’ve read the full thread and understand the Turbopack dev-mode issues (webpack config conflicts + dependency lock on Next 14). I’d like to take a fresh run at it: I’ll reproduce the errors, audit any webpack customisation, test both Next 14 & 15, and open a PR (or draft) with everything I find.
May I go ahead?

**Comment by imApoorva36** [comment]
Hi @JeelRajodiya 
I’ve fixed the Turbopack issues in #205.
The root cause was client-only libraries (Chakra/Emotion) being imported inside Server Components, which Turbopack enforces strictly. I’ve corrected the server/client boundaries and adjusted the config so Turbopack and the existing setup coexist cleanly on Next 14.
next dev --turbo now runs without hydration errors or build crashes.
Would love for you to review the PR when you find time, thanks!
